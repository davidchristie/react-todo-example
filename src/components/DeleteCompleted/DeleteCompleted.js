import PropTypes from 'prop-types'
import React from 'react'

import withItems from '../../enhancers/withItems'
import Button from '../Button'

export const DeleteCompleted = props => {
  return props.items.some(item => item.completed)
    ? (
      <Button className='DeleteCompleted' onClick={props.deleteCompleted}>
        Remove Completed
      </Button>
    )
    : null
}

DeleteCompleted.displayName = 'DeleteCompleted'

DeleteCompleted.propTypes = {
  deleteCompleted: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default withItems(DeleteCompleted)
