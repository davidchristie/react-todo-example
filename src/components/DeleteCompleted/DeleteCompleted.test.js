import { shallow } from 'enzyme'
import React from 'react'

import { DeleteCompleted } from './DeleteCompleted'

it('does not display if no items are completed', () => {
  const props = {
    items: [
      {
        completed: false,
        text: 'Test item 1'
      },
      {
        completed: false,
        text: 'Test item 2'
      }
    ]
  }
  expect(shallow(<DeleteCompleted {...props} />)).toMatchSnapshot()
})

it('does display if an item has been completed', () => {
  const props = {
    items: [
      {
        completed: false,
        text: 'Test item 1'
      },
      {
        completed: true,
        text: 'Test item 2'
      }
    ]
  }
  expect(shallow(<DeleteCompleted {...props} />)).toMatchSnapshot()
})
