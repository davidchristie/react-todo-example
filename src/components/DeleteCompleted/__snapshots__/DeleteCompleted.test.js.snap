// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does display if an item has been completed 1`] = `
<Button
  className="DeleteCompleted"
>
  Remove Completed
</Button>
`;

exports[`does not display if no items are completed 1`] = `""`;
